optimum/intel/utils/modeling.py

```diff
@@ -0,0 +1,337 @@
+
+                inputs["past_key_values"] = past_key_values
+
+            inputs["input_ids"] = input_ids
+
+            outputs = self.model(**inputs)
+            return CausalLMOutputWithPast(logits=outputs[0], past_key_values=outputs[1] if self.use_cache else None)
+        else:
+            return self.model.forward(
+                input_ids=input_ids,
+                past_key_values=past_key_values,
+                attention_mask=attention_mask,
+                **kwargs,
+            )
+
+    @classmethod
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        revision: Optional[Union[str, None]] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        file_name: Optional[str] = WEIGHTS_NAME,
+        local_files_only: bool = False,
+        use_cache: bool = True,
+        **kwargs,
+    ):
+        to_torchscript = kwargs.pop("torchscript", False)
+        if config.torchscript:
+            # Load the model from local directory
+            if os.path.isdir(model_id):
+                file_name = os.path.join(model_id, file_name)
+                model = cls.load_model(file_name)
+            # Download the model from the hub
+            else:
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+                model = cls.load_model(model_cache_path)
+
+            # Only for IPEX, IPEX jit model need 2 iterations to convert model to int8 model
+            onnx_config_class = TasksManager.get_exporter_config_constructor(
+                model_type=config.model_type, exporter="onnx", task=cls.export_feature
+            )
+            onnx_config = onnx_config_class(config, use_past=use_cache)
+            model_inputs = onnx_config.generate_dummy_inputs(framework="pt")
+            for i in range(2):
+                model(**model_inputs)
+        elif not to_torchscript:
+            task = cls.export_feature
+            subfolder = kwargs.pop("subfolder", "")
+            model_kwargs = {
+                "revision": revision,
+                "use_auth_token": use_auth_token,
+                "cache_dir": cache_dir,
+                "subfolder": subfolder,
+                "local_files_only": local_files_only,
+                "force_download": force_download,
+            }
+            model = TasksManager.get_model_from_task(task, model_id, **model_kwargs)
+            model_inputs = None
+        else:
+            return cls._from_transformers(

```

Comment:
I would rather keep the `export` parameter instead of `torchscript` to keep consistency accross all the `OptimizedModel`. To make it more explicit, why not change the class name instead ? To something like `TorchScriptModelForCausalLM` ?

Reply:
I have retained the export parameter, but I don't think the semantics of export match the tracing jit model very well, so I added the torchscript parameter. It's ok for me to change the class name, TorchScriptModelForCausalLM is OK.

Reply:
How about TorchScriptModelForGeneration? because not all generation model is CausalLM, like gpt2 is GPT2LMHeadModel